framework/src/parser/Syntax.C

```diff
@@ -228,3 +238,17 @@ Syntax::verifyMooseObjectTask(const std::string & base, const std::string & task
 
   return false;
 }
+
+std::pair<std::string, int>
+Syntax::getLineInfo(const std::string & syntax,
+                    const std::string & action,
+                    const std::string & task) const
+{
+  auto iters = _syntax_to_line.equal_range(syntax);
+  if (iters.first != iters.second)
+    for (auto && it = iters.first; it != iters.second; ++it)
+      if (std::get<0>(it->second) == action && std::get<1>(it->second) == task)
+        return std::make_pair(std::get<2>(it->second), std::get<3>(it->second));
```

Comment:
In `Syntax` we store a tuple, so these are just getting access to each value.
I think another structure would be good here too, especially given how this map is currently accessed.